### PR TITLE
Instrument Greenwave - Step 4 - UMB Messaging

### PR DESCRIPTION
**Pull request by xpokor85**
PVCORE-3883

There were some problems with the previous investigation and `consumers` are outdated so I instrumented also `listeners`.

As part of the ticket, I added a tutorial on manual checking the UMB messages in `functional-tests/manual_testing`.

### Current state (no incoming tracker):
```json
{
"correlationId" : "b5b1feff-0255-4461-9cc2-97250c0b1595",
"op" : "message-delivered",
"destination" : "topic://VirtualTopic.eng.greenwave.decision.update",
"originalDestination" : "null",
"username" : null,
"roles" : [ ],
"clientIp" : "tcp://10.89.0.10:35364",
"clientId" : "ID:99dd7cebb698-36971-1701083075287-3:1059",
"transportConnector" : "stomp",
"messageProperties" : {
"summary" : "1 of 1 required test results missing",
"subject_identifier" : "avahi",
"product_version" : "rhel-8",
"decision_context" : "osci_compose_gate_modules",
"subject_type" : "redhat-module",
"policies_satisfied" : "false"
},
"messageType" : null,
"messageExpirationTime" : 1701271319167,
"brokerPath" : null,
"brokerInTime" : 1701184919167,
"brokerOutTime" : 0,
"messageSize" : 2374,
"loggingBrokerName" : "umb"
}
```

### Desired state (incoming tracker is mocked):
```json
{
"correlationId" : "d38906d0-562b-4a23-ac94-192a8a93cd87",
 "op" : "message-delivered",
"destination" : "topic://VirtualTopic.eng.greenwave.decision.update",
"originalDestination" : "null",
"username" : null,
 "roles" : [ ],
 "clientIp" : "tcp://10.89.0.10:46358",
"clientId" : "ID:99dd7cebb698-36971-1701083075287-3:1064",
"transportConnector" : "stomp",
"messageProperties" : {
"summary" : "1 of 1 required test results missing",
"subject_identifier" : "avahi",
"product_version" : "rhel-8",
"trace_id" : "225655793668338684624371420195539389047",
"decision_context" : "osci_compose_gate_modules",
"subject_type" : "redhat-module",
"span_id" : "15680730273533578502",
"traceparent" : "00-a9c3b99a95cc045e573e163c3ac80a77-d99d251a8caecd06-01",
"policies_satisfied" : "false"
},
"messageType" : null,
"messageExpirationTime" : 1701271531015,
"brokerPath" : null,
"brokerInTime" : 1701185131015,
"brokerOutTime" : 0,
"messageSize" : 2603,
"loggingBrokerName" : "umb"
}
```

I added the "traceparent" to the headers also as it could be useful in total integration and it does include track and span ids.

### CODE REVIEW

**Comment by xpokor85** [comment]
The merge with the master created way more changes than it should.

**Comment by mike-kingsbury** [comment]
@xpokor85 I think you want the MR to merge to observability_main....  I believe that's the cause of the increase in changes...

**Comment by xpokor85** [comment]
Yeah, you are correct. Fixed my mistake

File: functional-tests/manual_testing/send_umb_message.py
[the remark this answers is not in the document]
Comment (pacho-rh) [reply functional-tests/manual_testing/send_umb_message.py:1]: At the moment, we won't automate this test. The test with UMB involves checking the UMB container logs to see if it received the UMB message generated by this test script which can be checked manually for now (similar to how we are check Jaegar manually).

My comment still stands regarding how Greenwave devs would feel about having this manual testing script here but I'll leave it to them to comment on that. If they have an issue with it, we can move these manual testing files elsewhere such as our Greenwave instrumentation documentation.

**Comment by pacho-rh** [comment]
LGTM

**Review by apetryla — APPROVED** [review]

File: greenwave/consumers/consumer.py [hunk 1]
```diff
@@ -108,6 +114,12 @@ def _inc(self, messaging_counter):
 
     def _publish_decision_update_fedora_messaging(self, decision):
         try:
+            TraceContextTextMapPropagator().inject(decision, self.context)
+            span = get_current_span(context=self.context).get_span_context()
```
[the remark this answers is not in the document]
Comment (xpokor85) [reply greenwave/consumers/consumer.py]: Addressed in consumers and listeners.

**Pull request merged**